tests/test_workflow_runner.py

```diff
@@ -178,6 +182,109 @@ def test_process_container_config_boutiques_subcommand(config: Config, tmp_path:
     )
 
 
+def test_check_tar_conditions_no_tracker_config(config: Config, tmp_path: Path):
+    runner = PipelineRunner(
+        dpath_root=tmp_path / "my_dataset",
+        pipeline_name="dummy_pipeline",
+        pipeline_version="1.0.0",
+        tar=True,
+    )
+    runner.config = config
+    runner.pipeline_step_config.TRACKER_CONFIG_FILE = None
+    with pytest.raises(
+        RuntimeError, match="Tarring requested but is no tracker config file"
+    ):
+        runner._check_tar_conditions()
+
+
+def test_check_tar_conditions_no_dir(config: Config, tmp_path: Path):
+    runner = PipelineRunner(
+        dpath_root=tmp_path / "my_dataset",
+        pipeline_name="dummy_pipeline",
+        pipeline_version="1.0.0",
+        tar=True,
+    )
+    runner.config = config
+    runner.pipeline_step_config.TRACKER_CONFIG_FILE = tmp_path  # not used
+    runner.tracker_config = TrackerConfig(
+        PATHS=[tmp_path], PARTICIPANT_SESSION_DIR=None
+    )
+    with pytest.raises(
+        RuntimeError,
+        match="Tarring requested but no participant-session directory specified",
+    ):
+        runner._check_tar_conditions()
+
+
+def test_check_tar_conditions_no_tar(config: Config, tmp_path: Path):
+    runner = PipelineRunner(
+        dpath_root=tmp_path / "my_dataset",
+        pipeline_name="dummy_pipeline",
+        pipeline_version="1.0.0",
+        tar=False,
+    )
+    runner.config = config
+    runner._check_tar_conditions()
+
+
+def test_tar_directory(tmp_path: Path):
+    # create dummy files to tar
+    dpath_to_tar = tmp_path / "my_data"
+    fpaths_to_tar = [
+        dpath_to_tar / "dir1" / "file1.txt",
+        dpath_to_tar / "file2.txt",
+    ]
+    for fpath in fpaths_to_tar:
+        fpath.parent.mkdir(parents=True, exist_ok=True)
+        fpath.touch()
+
+    runner = PipelineRunner(
+        dpath_root=tmp_path / "my_dataset",
+        pipeline_name="dummy_pipeline",
+        pipeline_version="1.0.0",
+    )
+    fpath_tarred = runner.tar_directory(dpath_to_tar)
+
+    assert fpath_tarred == dpath_to_tar.with_suffix(".tar")
+    assert fpath_tarred.exists()
+    assert fpath_tarred.is_file()
+
+    with tarfile.open(fpath_tarred, "r") as tar:
+        tarred_files = [tarred for tarred in tar.getmembers() if tarred.isfile()]
+        assert len(tarred_files) == len(fpaths_to_tar)
+        for member in tarred_files:
+            assert tmp_path / member.name in fpaths_to_tar
```

Comment:
I think this should be reversed to ensure the source files are in the TAR. As of now, if a source file isn't in the TAR, then the test still pass.